### PR TITLE
Added PHP 5.5. compatibility

### PR DESCRIPTION
**Pull request by OndraM**
PHP 5.5 is not activelly supported, though it is still in security support term (https://secure.php.net/supported-versions.php) and still widely used.

As the dependency on PHPUnit 5 is the only reason for PHP 5.6 (and the tests are also working properly with PHP 4.8) I propose adding support also for PHP 5.5 - at least for now.

Ref #2.

### CODE REVIEW

**Comment by snapshotpl** [comment]
+1

File: composer.json [hunk 1]
```diff
@@ -11,12 +11,12 @@
...
     ],
     "require": {
-        "php": ">=5.6",
+        "php": ">=5.5",
         "ramsey/uuid": "^3.0",
         "moontoast/math": "^1.1"
     },
     "require-dev": {
-        "phpunit/phpunit": "^5.0"
+        "phpunit/phpunit": ">=4.8"
```
Comment (vinkla) [inline composer.json:19]: This should be `^4.8 || ^5.0`

File: composer.json [hunk 2]
```diff
@@ -11,12 +11,12 @@
         }
     ],
     "require": {
-        "php": ">=5.6",
+        "php": ">=5.5",
```
Comment (vinkla) [inline composer.json:14]: This should be `^5.5.9 || ^7.0`

**Comment by vinkla** [comment]
:-1: since `5.5` is only four months way of being discontinued I don't think it is worth it.

**Comment by pascaldevink** [comment]
Nope, it's not worth it and I won't merge any request that adds support for it.

**Comment by TomasVotruba** [comment]
@OndraM Looks like code of this package is PHP 5.5 compatible, co you could [just fake the platform]([platform]%28https://getcomposer.org/doc/06-config.md#platform%29) in your `composer.json` for now.

**Comment by OndraM** [comment]
@TomasVotruba Yes, but this is slightly dangerous, as it would impose also other packages to install their PHP 5.6 compatible version.

**Comment by TomasVotruba** [comment]
@OndraM Sure, that might happen. In that case add your own branch of fork to `composer.json`. That's the safest and fastest way ATM.

**Comment by OndraM** [comment]
@pascaldevink I understand and this is your call, though I would decide differently. 

For anyone interested I've prepared [PHP 5.5 compatible fork](https://github.com/OndraM/shortuuid) and published it to [Packagist](https://packagist.org/packages/ondram/shortuuid-php55) to to make it easily installable via Composer.
I do not intend to maintain it in longer term, I hope we will switch to PHP 5.6+ asap.

`composer require ondram/shortuuid-php55`

**Comment by TomasVotruba** [comment]
@OndraM Awesome! Great job

**Pull request closed without merging**